api/src/Command/Cron.php

```diff
@@ -140,35 +142,43 @@ public function runTask(): bool
         $last_task_timestamp = $this->system->get('last_task_timestamp');
         $last_task_name = $this->system->get('last_task_name');
 
-        // if a task is running and it's not older than MAX_TASK_LOCK_DURATION, do nothing
+        // if a task is running and it's not older than MAX_TASK_LOCK_DURATION, only run our "always" tasks
+        $onlyRunAlwaysTasks = false;
         if ($task_running && isset($last_task_timestamp) && $last_task_timestamp > time() - intval($this->params->get('max_task_lock_duration'))) {
             $this->logger->notice($last_task_name.' is already running since '.(time() - $last_task_timestamp).'s');
 
-            return false;
+            $onlyRunAlwaysTasks = true;
         }
 
-        // if a task is running but it's old, it's probably a lockup.
-        // continue but log it
-        if (isset($last_task_timestamp) && $last_task_timestamp < time() - intval($this->params->get('max_task_lock_duration'))) {
-            $this->logger->notice('Removing old task lock');
-        }
+        // Don't count the always-only task run as the last task when checking for duplicates
+        if (!$onlyRunAlwaysTasks) {
+            // if a task is running but it's old, it's probably a lockup.
+            // continue but log it
+            if (isset($last_task_timestamp) && $last_task_timestamp < time() - intval($this->params->get('max_task_lock_duration'))) {
+                $this->logger->notice('Removing old task lock');
+            }
 
-        $this->system->set('task_running', true);
+            $this->system->set('task_running', true);
 
-        // get idle hours
-        if (null !== $this->params->get('idle_hours')) {
-            $idle_hours = explode('-', $this->params->get('idle_hours'));
-            $idle_hours[0] = intval($idle_hours[0]);
-            $idle_hours[1] = intval($idle_hours[1]);
-        } else {
-            $idle_hours = [0, 24];
+            // get idle hours
+            if (null !== $this->params->get('idle_hours')) {
+                $idle_hours = explode('-', $this->params->get('idle_hours'));
+                $idle_hours[0] = intval($idle_hours[0]);
+                $idle_hours[1] = intval($idle_hours[1]);
+            }
         }
 
         foreach ($this->tasks as $task) {
+
+            // If we are only doing "always" tasks, skip others
+            if (!isset($task['type'])
+                || $onlyRunAlwaysTasks && 'always' !== $task['type']) {
```

Comment:
I just learned that `&&` has higher precedence than `||` in php so your condition works. But it's maybe more readable if you add a pair of parenthesis around `$onlyRunAlwaysTasks && 'always' !== $task['type']` ?

Reply:
Haha I was going to do that but I knew `&&` has higher precedence in PHP (I think this was true for most languages?) so I thought you might consider it unnecessary.

I've used the style from something further down so it should be consistent now.